src/inmanta/agent/in_process_executor.py

```diff
@@ -286,15 +261,23 @@ async def dry_run(
             with self._cache:
                 started = datetime.datetime.now().astimezone()
                 try:
-                    resource_obj: Resource | None = await self.deserialize(resource, const.ResourceAction.dryrun)
-                except Exception:
+                    resource_obj: Resource = Resource.deserialize(resource.attributes)
+                except Exception as e:
+                    msg = data.LogLine.log(
```

Comment:
I think you have to flush this to the logger (it has a method for that)

Reply:
Added the `_log_deserialization_error` method that I believe does this